model/src/main/java/com/cloudera/nav/sdk/model/IdAttrs.java

```diff
@@ -0,0 +1,81 @@
+// (c) Copyright 2015 Cloudera, Inc. All rights reserved.
+package com.cloudera.nav.sdk.model;
+
+import com.cloudera.nav.sdk.model.annotations.MProperty;
+import com.cloudera.nav.sdk.model.entities.EntityType;
+import com.cloudera.nav.sdk.model.entities.TagChangeSet;
+import com.cloudera.nav.sdk.model.entities.UDPChangeSet;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.Instant;
+
+/**
+ * Used for deserializing custom entities. We use this as an intermediary
+ * so that we can handle the changesets for tags/UDPs
+ */
+public class IdAttrs {
```

Comment:
if i understand correctly you mean putting a Map in the Entity object instead of having an Object here? We essentially want to keep this structure because this is what an ep1/ep2 attributes looks like for the Relation object. And also it makes it easy for users to just say hdfsDir.setFileSystemPath()